truffle/com.oracle.truffle.api.instrumentation/src/com/oracle/truffle/api/instrumentation/SourceSectionFilter.java

```diff
@@ -362,7 +362,7 @@ public SourceSectionFilter build() {
         }
 
         private void verifyNotNull(Object[] values) {
-            if (values == null) {
+            if (values == null || values.length == 0) {
                 throw new IllegalArgumentException("Given arguments must not be null.");
```

Comment:
Should be a different error message for `value.length == 0`. Also, the method name does not indicate that this check is done.